### PR TITLE
Upgrade unicorn to play 2.5, introduce DI, remove Play.current #82 #75

### PR DESCRIPTION
**Pull request by liosedhel**
Upgrade unicorn to play 2.5, introduce DI, remove Play.current #82 #75

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -17,13 +17,13 @@ object Dependencies {
   def core(scalaVersion: String) = mainCore(scalaVersion) ++ testCore
 
   val mainPlay = Seq(
-    "com.typesafe.play" %% "play-slick" % "1.1.0"
+    "com.typesafe.play" %% "play-slick" % "2.0.0"
   )
 
   val testPlay = Seq(
-    "com.typesafe.play" %% "play-test" % "2.4.3" % "test"
+    "com.typesafe.play" %% "play-test" % "2.5.2" % "test"
```
Comment (mgilbertnz) [inline project/Dependencies.scala]: Can we bump this to 2.5.3 while we're here please?

Comment (liosedhel) [inline project/Dependencies.scala]: Yes, done.

File: project/Settings.scala [hunk 2]
```diff
@@ -14,10 +13,10 @@ object Settings {
   // settings for ALL modules, including parent
   val core = Seq(
     organization := "org.virtuslab",
-    scalaVersion := "2.11.6",
-    crossScalaVersions := Seq("2.10.4", scalaVersion.value),
-    ReleaseKeys.crossBuild := true,
-    ReleaseKeys.publishArtifactsAction := PgpKeys.publishSigned.value,
+    scalaVersion := "2.11.7",
+    crossScalaVersions := Seq(scalaVersion.value),
```
Comment (Kwestor) [inline project/Settings.scala]: We don't need `crossScalaVersions` and `crossBuild` anymore.

File: version.sbt [hunk 3]
```diff
@@ -1 +1 @@
-version in ThisBuild := "1.0.1-SNAPSHOT"
\ No newline at end of file
+version in ThisBuild := "1.1.0-SNAPSHOT"
```
Comment (Kwestor) [inline version.sbt]: Was this updated by a release plugin?

**Pull request closed without merging**